clustered/client/src/main/java/org/ehcache/clustered/client/internal/service/ConnectionState.java

```diff
@@ -149,7 +149,11 @@ private void reconnect() {
   }
 
   private void connect() throws ConnectionException {
-    clusterConnection = connectionSource.connect(connectionProperties);
+    try {
+      clusterConnection = connectionSource.connect(connectionProperties);
+    } catch (ConnectionClosedException closed) {
```

Comment:
I think I'd prefer it if we just caught this in the callers instead. That should only be a small number of places. Should be able to use a multi-catch and treat it the same as the existing `ConnectionException` catches.